core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -64,18 +65,89 @@ public class OAuth2Manager extends RefreshingAuthManager {
   private AuthSessionCache sessionCache;
 
   public OAuth2Manager(String managerName) {
-    super(managerName + "-token-refresh");
+    super(managerName);
     this.name = managerName;
   }
 
   @Override
+  public OAuth2Manager withClient(RESTClient restClient) {
+    Preconditions.checkNotNull(restClient, "invalid restClient: null");
```

Comment:
this should probably be a IAE to indicate that the argument passed cannot be null. Also I would update the error msg to `Invalid REST client: null`